src/bridge/bridge-unix-sockets.cpp

```diff
@@ -73,98 +74,157 @@ BasicLocalClient client{};
 inline constexpr int BUFFER_SIZE = 1024;
 using ByteBuffer = std::array<uint8_t, BUFFER_SIZE>;
 ByteBuffer byteBuffer;
+}
+
+SlimeVRDriver::Bridge::Bridge(VRDriver *i_pDriver)
+        : m_pDriver(i_pDriver), m_pPipe(nullptr), m_pBuffer(nullptr),
+          m_eState(EState::BRIDGE_DISCONNECTED), m_bStop(false) {
+}
+
+SlimeVRDriver::Bridge::~Bridge() {
+    this->stop();
+}
+
+void SlimeVRDriver::Bridge::run() {
+
+    messages::ProtobufMessage oMsg;
+
+    while (!this->m_bStop) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(3));
```

Comment:
Ideally, we'd wait for the OS to notify us that there's a message incoming, or we have a message in the queue that needs to be sent instead of sleeping for \<arbitrary amount of time\>.

Reply:
I'm talking about using select or WaitForMultipleObjects instead of sleeping, meaning the OS can wake us up as soon as either there's an incoming message, or another thread sends a notification that there's an outgoing message, and we don't need to periodically wake-up to poll the outgoing queue.

This would still be applicable to a TCP based implementation, unless you had the other thread just send the message directly. (But that optimization would also apply to this implementation, so *eh*)

This doesn't need to be done right now, so I'll resolve the conversation after adding a TODO.

Reply:
```suggestion
        // TODO: use select, or a timeout, or just send the messages from another thread.
        std::this_thread::sleep_for(std::chrono::milliseconds(3));
```